USING_ELEKTRA.md

```diff
@@ -0,0 +1,165 @@
+#> cyan
+```
+The result comes from the set `default` metadata which is saved for the key. 
+You can see all metadata by calling `kdb lsmeta`:
+```sh
+kdb lsmeta '/sw/lcdproc/lcdd/#0/current/curses/background'
+#> check/enum
+#> check/type
+#> default
+#> description
+#> type
+```
+Here we can als see `check/enum` which is used by Elektra to assert
+for correct configuration settings. You can query for the values by calling
+`kdb getmeta`
+```sh
+kdb getmeta '/sw/lcdproc/lcdd/#0/current/curses/background' 'check/enum'
+#> red, black, green, yellow, blue, magenta, cyan, white
+```
+If you try to set the value of the background to an invalid value, Elektra
+will raise an error and prevent a possible misconfiguration:
+```sh
+kdb set '/sw/lcdproc/lcdd/#0/current/curses/background' purple
+#> Using name user/sw/lcdproc/lcdd/#0/current/curses/background
+#> The command kdb set failed while accessing the key database with the info:
+#> Sorry, the error (#121) occurred ;(
+#> Description: validation failed
+#> Reason: Validation of key "user/sw/lcdproc/lcdd/#0/current/curses/background" with string "purple" failed.
+#> Ingroup: plugin
+#> Module: enum
+#> At: <path>/src/plugins/enum/enum.c:218
+#> Mountpoint: user/sw/lcdproc/lcdd/#0/current
+#> Configfile: <path>/.config/LCDd.conf.17145:1549993505.561040.tmp
+```
+
+### Kdb editor
+
+Another sophisticated possibility to see available configuration settings and edit them
+is to use the `kdb editor`. If you call `kdb editor <key>`, a local editor will
+be opened and all keys under the given `<key>` including all metadata will be shown.
+Depending on your default storage plugin you have chosen in Elektra the output may differ.
+The default storage plugin is the `dump` plugin and the output looks like this:
+```sh
+kdb editor '/sw/lcdproc/lcdd/#0/current/curses/background'
+##Editor
+#> kdbOpen 1
+#> ksNew 1
+#> keyNew 50 1
+#> spec/sw/lcdproc/lcdd/#0/current/curses/background^@^@
+#> keyMeta 11 54
+#> check/enum^@red, black, green, yellow, blue, magenta, cyan, white^@
+#> keyMeta 11 5
+#> check/type^@enum^@
+#> keyMeta 8 5
+#> default^@cyan^@
+#> keyMeta 12 41
+#> description^@background color when "backlight" is off^@
+#> keyMeta 5 5
+#> type^@enum^@
+#> keyEnd
+#> ksEnd
+```
+This is the actual format how the `dump` plugin saves data in a file. You
+can see all metadata and its associated meta values. But if you look closely, 
+these values are all under the `spec` namespace and changing anything here will not make a difference.
+Now lets see what happens when we set the background differently:
+```sh
+kdb set '/sw/lcdproc/lcdd/#0/current/curses/background' green
+#> Using name user/sw/lcdproc/lcdd/#0/current/curses/background
+#> Create a new key user/sw/lcdproc/lcdd/#0/current/curses/background with string "green"
+```
+A new key is created and associated with all metadata from the `spec` namespace:
+```sh
+kdb editor '/sw/lcdproc/lcdd/#0/current/curses/background'
```

Comment:
You should probably mention that you can choose the storage plugin used for editing.

The following chooses the `ni` plugin for example (which is much nicer than `dump`):
`kdb editor '/sw/lcdproc/lcdd/#0/current/curses/background' ni`